### ISSUE TITLE
Community ?

### ISSUE THREAD

**Issue by rapha8l** [geophile/marcel#11]
Hi,

Thanks for marcel, though I think it'd deserve some more acknoledgment
Maybe you should have an irc/matrix/discord/* chan somewhere ?
have a nice day

**Comment by geophile** [comment]
Thank you! 

I am quite ancient (65), so while I know how to write software, I don't know how to create a community around it. That really wasn't a thing when I was younger.

I haven't create a discussion forum anywhere because marcel has no community that I am aware of. Or maybe I need to create the forum to build the community? I really don't know. Anyway, if you have suggestions on how to proceed, I would welcome them. Suppose I set up a discord channel. What next? How do I let people who might care know about it? 

Seriously, I am completely ignorant on these issues.

**Comment by rapha8l** [comment]
Hi, 
I'm not so young too, almost 50 :)
but well opening an irc chan or discord one, advertise it here, and maybe on other (You've been on the nu discord if i remember)
there is also the option of a reddit page
But the most simple is opening a chan somewhere, and let it live (or die) if you don't want to spend too much time on it.
but for people beginning using marcel, having somewhere to ask questions would be cool

**Comment by geophile** [comment]
I was on the nu discord briefly. 

Hadn't considered reddit, but that's a good idea too. I will start setting up discord and reddit.

By the way, I've had very little feedback on marcel. So far, everything is based on my requirements from a system I worked on (resulting in https://github.com/osh), and then my thoughts on how to turn osh into more of a full-fledged shell. So I would love another point of view. Any comments on: what you like; what you dislike; what it needs -- that would be really helpful.

**Comment by rapha8l** [comment]
Great !
I have to try a bit more marcel to be honest, I wanted to use to get more into python too, but time is short :)
My opinion is to have a real nice user experience (like fish or nu, for instance) and maybe bring more interactive features.
But I'm biaised, I've always thought cli/tui have a real interesting future over gui for IHM (even FUI, futuristic user interfaces show more objects and text than GUI's)

**Comment by geophile** [comment]
Yeah, marcel is kind of useless without a little Python (at least for writing expressions). 

Nice UI: Do you mean tabular output like nu? What fish features are tyou thinking of?

I agree, there is always a place for a CLI/TUI. 

What is IHM?

**Comment by rapha8l** [comment]
IHM is human machine interface, the acronym is french sorry :)
fish has nice autocomplete, coloring and is easy to use by default
tabular output is an idea, i think there can be others
my idea is that something useful has te be pleasant to use and nice looking :)

**Comment by geophile** [comment]
Here's what marcel has:

- Autocomplete: Pretty similar to Linux shells, I think. Auto-complete for marcel commands, linux executables, filenames, flags. 
- Coloring: The default config file does not have coloring. But it is easy to specify. Easier than bash anyway, where you have to know ASCII escape codes and shell extensions. Here is my config file (~/.marcel.py). All those COLOR_... entries, PROMPT and PROMPT_CONTINUATION specify colors, used in prompts and for colorizing command output.
[m.txt](https://github.com/geophile/marcel/files/8791749/m.txt)

**Comment by rapha8l** [comment]
thanks

**Comment by rapha8l** [comment]
oh, and if I can suggest, using  $XDG_CONFIG_HOME for dotfiles would be better imho

**Comment by rapha8l** [comment]
you can put all in ~/.config/marcel with something like rc.py or conf.py (or anything else) that could source subdirectories likes ~/.config/marcel/plugins, ~/.config/marcel/themes or anything that comes to your idea :)

**Comment by geophile** [comment]
Commit e47a40b3515e7853173ce68ce6bb7bc564495440 moves the startup file to $XDG_CONFIG_HOME/startup.py, or $HOME/.config/marcel/startup.py, as discussed above.

**Comment by rapha8l** [comment]
nice, thanks

**Comment by geophile** [comment]
A bit more work on this. The marcel history file (log of commands, much like .bash_history) is now kept in $XDG_DATA_HOME/marcel/history.

**Comment by rapha8l** [comment]
great !
